core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -19,162 +19,237 @@
-      controller: ['$scope', '$modalInstance', function($scope, $modalInstance) {
-        $scope.newThreadSubject = '';
-        $scope.newThreadText = '';
-
-        $scope.create = function(newThreadSubject, newThreadText) {
-          if (!newThreadSubject) {
-            warningsData.addWarning('Please specify a thread subject.');
-            return;
+    $rootScope.loadingMessage = 'Loading';
+    $scope.tmpMessage = {
+      status: null,
+      text: ''
+    };
+
+    var _resetTmpMessageFields = function() {
+      $scope.tmpMessage.status = $scope.activeThread ?
+        $scope.activeThread.status : null;
+      $scope.tmpMessage.text = '';
+    };
+
+    $scope.clearActiveThread = function() {
+      $scope.activeThread = null;
+      _resetTmpMessageFields();
+    };
+
+    $scope.showCreateThreadModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorFeedbackCreateThread',
+        backdrop: true,
+        resolve: {},
+        controller: ['$scope', '$modalInstance', function(
+          $scope, $modalInstance) {
+          $scope.newThreadSubject = '';
+          $scope.newThreadText = '';
+
+          $scope.create = function(newThreadSubject, newThreadText) {
+            if (!newThreadSubject) {
+              warningsData.addWarning('Please specify a thread subject.');
+              return;
+            }
+            if (!newThreadText) {
+              warningsData.addWarning('Please specify a message.');
+              return;
+            }
+
+            $modalInstance.close({
+              newThreadSubject: newThreadSubject,
+              newThreadText: newThreadText
+            });
+          };
+
+          $scope.cancel = function() {
+            $modalInstance.dismiss('cancel');
+          };
+        }]
+      }).result.then(function(result) {
+        threadDataService.createNewThread(
+          result.newThreadSubject, result.newThreadText, function() {
+            $scope.clearActiveThread();
+          });
+      });
+    };
+
+    var _isSuggestionOpen = function() {
+      return $scope.activeThread.status === 'open';
+    };
+
+    var _isSuggestionCurrent = function() {
+      var activeThread = $scope.activeThread;
+      var suggestionExpVersion = activeThread.suggestion.exploration_version;
+      return explorationData.data.version === suggestionExpVersion;
+    };
+
+    var _hasUnsavedChanges = function() {
+      return (changeListService.getChangeList().length > 0);
+    };
+
+    $scope.viewSuggestionBtnType = function() {
+      return (_isSuggestionOpen() && _isSuggestionCurrent() &&
+              !_hasUnsavedChanges() ? 'primary' : 'default');
+    };
+
+    // TODO(Allan): Implement ability to edit suggestions before applying.
+    $scope.showSuggestionModal = function() {
+      $modal.open({
+        templateUrl: 'modals/editorViewSuggestion',
+        backdrop: true,
+        size: 'lg',
+        resolve: {
+          isSuggestionOpen: function() {
```

Comment:
Suggest: suggestionIsOpen, suggestionIsCurrent, unsavedChangesExist to indicate that these are nouns and not functions (whose names generally begin with verbs).